### ISSUE TITLE
use-any reports incorrectly that I should use any

### ISSUE THREAD

**Issue by heynemann** [mgechev/revive#662]
**Describe the bug**
use-any rule is incorrectly reporting an error

**To Reproduce**

Using this code:

```
type MyInterface interface {

}
```

It reports that I should use any, but that's not correct as I just declared an empty named interface. It's not like map[string]interface{} where I actually should use any.

**Expected behavior**

It should report only for incorrect uses of empty interfaces, not on type declarations.

**Comment by chavacava** [comment]
Hi @heynemann, thanks for filling the issue.
The GO 1.18 release note says "_The new [predeclared identifier](https://tip.golang.org/ref/spec#Predeclared_identifiers) any is an alias for the empty interface. It may be used instead of interface{}._"
Nothing indicates it can not be used in type declarations.
Looking into the massive PR in the GO repo where `interface{}` was replaced by `any` you can find [examples of types declarations where `interface{}` was replaced by `any`](https://github.com/golang/go/commit/2580d0e08d5e9f979b943758d3c49877fb2324cb#diff-eab68ada2b589d786862a7ba9c02a24e65fae68df07e05793f93a7f47b076297)

**Comment by heynemann** [comment]
I get what you are saying and the distinction is really difficult to do in a linter, but those are type aliases. I really shouldn't have an interface with no methods (as it does not make sense), and would rather have that as a linter, but that won't work either as you won't be able to differentiate between:

```go
type MyInterface interface {

}
```

and 

```go
type MyTypeAlias interface{}
```

The first would be used to do `variable.(MyInterface)` casting. The second is to be used in declaring variables and fields.

Closing as this is a problem with my code. Thanks for looking into it. But the code in Go's source that uses any uses it for type aliases (as they don't have any empty interfaces otherwise, I guess).

**Comment by chavacava** [comment]
I do not get the point.
Do you mean that there is a difference between 
```go
type too  interface{}
```
and 
```go
type too  interface{

}
```
?
It is a type definition with an empty interface in both cases.. isn't it?

> the code in Go's source that uses any uses it for type aliases (as they don't have any empty interfaces otherwise, I guess).

The examples I've pointed in the `src/crypto/crypto.go` of the GO PR are all type definitions. (alias definitions use `=` between the type name and the type)
